### PR TITLE
Implement force_wrappers argument of register_extendr()

### PR DESCRIPTION
**Pull request by yutannihilation**
As suggested here: https://github.com/extendr/rextendr/issues/56#issuecomment-792186856

### CODE REVIEW

**Review by clauswilke — APPROVED** [review]
It's generally fine. I'm just not sure about the word "sample" in `sample_function_wrapper()` etc. What is it supposed to mean? Do you mean "example"?

**Comment by clauswilke** [comment]
The other thing I've been wondering is whether we should check for the presence of `src` and `src/entrypoint.c` and execute `register_wrappers()` only if they exist? As a way to check whether there actually is compiled Rust code for which we can retrieve wrappers.

**Comment by yutannihilation** [comment]
Thanks, I'll use "example".

File: R/register_extendr.R [hunk 1]
```diff
@@ -12,20 +12,45 @@
...
   pkg_name <- x$get("Package")
 
   if (!isTRUE(quiet)) {
     message(glue("Generating extendr wrapper functions for package: {pkg_name}"))
   }
 
+  entrypoint_c_file <- rprojroot::find_package_root_file("src", "entrypoint.c", path = ".")
+  if (!file.exists(entrypoint_c_file)) {
+    stop(
+      "src/entrypoint.c, which is necessary to make wrappers available to R sessions, doesn't exist.",
```
Comment (yutannihilation) [inline R/register_extendr.R:32]: Implemented the check on `src/entrypoint.c`. But I'm still thinking what the error message should be here...

File: R/register_extendr.R [hunk 2]
```diff
@@ -12,20 +12,45 @@
...
+  # If FALSE, execute make_wrappers() only when the package can be loaded.
+  if (isTRUE(force_wrappers)) {
+    tryCatch(
+      make_wrappers(pkg_name, pkg_name, outfile, use_symbols = TRUE, quiet = quiet),
+      error = function(...) {
+        warning(
+          "Generating the wrapper functions failed, so a minimal one is used instead",
+          call. = FALSE
+        )
+        make_sample_wrappers(pkg_name, outfile)
```
Comment (clauswilke) [inline R/register_extendr.R:50]: I think this needs to be updated here also.

File: R/register_extendr.R (hunk 1 above)
Comment (clauswilke) [reply R/register_extendr.R:32]: Maybe write: "Could not find `src/entrypoint.c`. Are you sure this package is using extendr Rust code?"

**Comment by clauswilke** [comment]
I think you should also add yourself to the authors list in `DESCRIPTION`.

File: R/register_extendr.R (hunk 1 above)
Comment (yutannihilation) [reply R/register_extendr.R:32]: Sounds good, thanks.

File: R/register_extendr.R (hunk 2 above)
Comment (yutannihilation) [reply R/register_extendr.R:50]: Oh, good catch...

**Comment by yutannihilation** [comment]
Merging now. It's an honor to join the authors list :)

**Pull request merged**